### ISSUE TITLE
Order Creation: Only first page of products can be added to order

### ISSUE THREAD

**Issue by rachelmcr** [woocommerce/woocommerce-android#6217]
**Describe the bug**

During order creation, I can only add the first page of products from the API to a new order. Subsequent pages of products aren't loaded in the product list on the Add product screen, and if I search for and select a product that isn't in the list it isn't added to the order.

**To Reproduce**
1. Make sure you have more than one page of products in your store.
2. In the app, go to the Orders tab and create a new order.
3. Select "Add product" and scroll down the product list. Notice that the entire product list doesn't load.
4. Search for a product that doesn't appear in the list, and select it to add it to the order.
5. On the new order screen, notice that the selected product isn't added to the order.

**Screenshots**

https://user-images.githubusercontent.com/8658164/162473178-d6d62bce-94d8-4635-83ea-c5e0fa3cb7a8.mp4

**Comment by nbradbury** [comment]
@rachelmcr So far I haven't been able to reproduce the lack of infinite scroll, and I tried with a fresh install and never visited the main product list. I have noticed, though, that the "Add product" list is fairly janky, and the "load more listener" is called more often than it is in the main product list, and the adapter is also loaded more often than in the main product list.

**Comment by nbradbury** [comment]
And of course right after I posted the above comment I was able to reproduce it :)

**Comment by nbradbury** [comment]
I'm still not sure why infinite scroll is so flaky, but I did uncover the problem with searched products not being added. Looking at the FluxC code, [handleSearchProductsCompleted](https://github.com/wordpress-mobile/WordPress-FluxC-Android/blob/23e8fe682e21e4e045cf365a06ca8be46f55d425/plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/store/WCProductStore.kt) is called when the search completes, and the list of matching products is _not_ being saved to the database. This mean when the order creation asks the db for the product, it's not there so it doesn't get added.

I was going to ask around why we don't save searched products, but git blame showed I wrote the code :)
